### PR TITLE
Re enable error-prone

### PR DESCRIPTION
**Pull request by fsamuel-bs**
**Goals (and why)**: Re enable error-prone.

**Implementation Description (bullets)**: Wrap error-prone syntax checks in a gradle property variable. Run checkstyle checks with that variable enabled, run other checks with that variable disabled.

**Concerns (what feedback would you like?)**: -

**Where should we start reviewing?**: -

**Priority (whenever / two weeks / yesterday)**: -

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/1924)

### CODE REVIEW

File: scripts/circle-ci/run-circle-tests.sh [hunk 1]
```diff
@@ -22,6 +22,8 @@ CONTAINER_5=(':atlasdb-ete-tests:longTest' ':lock-impl:check' ':atlasdb-dbkvs-te
 
 CONTAINER_6=(':atlasdb-ete-test-utils:check' ':atlasdb-cassandra:check' ':atlasdb-api:check' ':atlasdb-jepsen-tests:check' ':atlasdb-cli:check')
 
+CONTAINER_7=('checkstyle{Main,Test} -PenableErrorProne=true')
```
Comment (hsaraogi) [inline scripts/circle-ci/run-circle-tests.sh]: This should be `compileJava/compileTestJava`, checkStyle is run by the build task in other containers as well.

File: scripts/circle-ci/run-circle-tests.sh [hunk 2]
```diff
@@ -78,4 +80,5 @@ case $CIRCLE_NODE_INDEX in
     4) ./gradlew $BASE_GRADLE_ARGS ${CONTAINER_4[@]} ;;
     5) ./gradlew $BASE_GRADLE_ARGS ${CONTAINER_5[@]} ;;
     6) ./gradlew $BASE_GRADLE_ARGS ${CONTAINER_6[@]} -x :atlasdb-jepsen-tests:jepsenTest && checkDocsBuild ;;
+    7) ./gradlew $BASE_GRADLE_ARGS ${CONTAINER_7[@]} ;;
```
Comment (hsaraogi) [inline scripts/circle-ci/run-circle-tests.sh]: Please add container 7 to circle(internal and external). Let me know if I can help here.

File: gradle/baseline.gradle [hunk 3]
```diff
@@ -9,20 +9,26 @@ List<String> blacklistedBaselineProjects = [
         'atlasdb-rocksdb',
         'atlasdb-rocksdb-tests',
         'atlasdb-service-server',
-        'cassandra-partitioner',
+//        'cassandra-partitioner',
```
Comment (hsaraogi) [inline gradle/baseline.gradle]: Stray?

Comment (fsamuel-bs) [inline gradle/baseline.gradle]: Just testing if it will fail.

File: gradle/baseline.gradle [hunk 4]
```diff
@@ -17,12 +17,18 @@ List<String> blacklistedBaselineProjects = [
...
 
-    // We are not presently using errorprone because that swaps out the compiler to one based on the beta compiler
-    // for JDK 9, which has a known issue with varargs.
+    // ./gradlew compileJava compileTestJava -PenableErrorProne=true
+    if (enableErrorProne.toBoolean()) {
+        apply plugin: 'org.inferred.processors'  // installs the "processor" configuration needed for baseline-error-prone
+        apply plugin: 'com.palantir.baseline-error-prone'
+
+        configurations.errorprone {
+            resolutionStrategy { force('com.google.guava:guava:21.0') }
```
Comment (ashrayjain) [inline gradle/baseline.gradle]: out of curiousity, why are we forcing this?

Comment (fsamuel-bs) [inline gradle/baseline.gradle]: I'm not really sure. It's not documented by the error-prone docs, but seems to be a widespread practice regarding our internal usage. It was added on #1853, I'm just adding it back. Maybe @gsheasby has a better understanding of this.

**Pull request merged**